### ISSUE TITLE
Breaking change with v0.17

### ISSUE THREAD

**Issue by cesarandreu** [goatslacker/alt#383]
> You may now return from actions directly in order to dispatch, no need to call `this.dispatch`.

This is a breaking change for me :(. I'm using async functions, which return promises. After performing some async things, I'll dispatch something manually.

Example:

```js
class FooFormActions {
  async success ({ response, form, ...args }) {
    const body = await response.json()
    const normalized = getNormalizedBody(form, body)
    this.dispatch({ form, body, ...normalized, ...args })
  }
}
```

**Comment by goatslacker** [comment]
I'll wrap this in a isPromise and we can ignore those.

**Comment by pstoica** [comment]
Could this be opt-out? We're returning non-promise tokens in a couple of actions (they could really be promises, but it's too late for now). If a this.dispatch occurs, don't use the return value?

**Comment by pstoica** [comment]
Also, if an action were to invoke another action, does that still work fine?

**Comment by goatslacker** [comment]
No, that would dispatch. An action that just dispatches another action is kinda weird, just call the resulting action.

**Comment by goatslacker** [comment]
This should've been closed. Fixed in alt@0.17.1

**Comment by cesarandreu** [comment]
Thanks :D.
